pandas/core/strings/accessor.py

```diff
@@ -1540,10 +1556,21 @@ def replace(
         if case is None:
             case = True
 
-        result = self._data.array._str_replace(
-            pat, repl, n=n, case=case, flags=flags, regex=regex
-        )
-        return self._wrap_result(result)
+        if isinstance(pat, dict):
```

Comment:
Instead of duplicating code below, can you do:

```
if not isinstance(pat, dict):
    pat = {pat: repl}
```

Then you only need one code path.

Reply:
@rhshadrach Is this what you're thinking? Because I'm getting a bunch of PerformanceWarning errors

![image](https://github.com/pandas-dev/pandas/assets/45905457/d35d587c-d262-49ef-8451-35ad01bb9b19)
![image](https://github.com/pandas-dev/pandas/assets/45905457/4b4f9c8c-2a86-4c49-9f06-381474c2fddc)

Reply:
No, you have `pat = {pat: dict}` whereas my suggestion was `pat = {pat: repl}`